src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs

```diff
@@ -102,6 +103,21 @@
            (create-field name (display-value value)))
          log-map)])))
 
+(defn- failure-tree [failures]
+  [comps/Tree {:start-collapsed? false
+               :label "Failures"
+               :highlight-ends false
+               :data (prewalk
+                       (fn [elem]
+                         (cond
+                           ;; reorder map elements to message then causedBy
```

Comment:
I'm not sure, I'd have to take an in-depth look at comps/Tree and what this data looks like.  I'm guessing that this entire prewalk is unnecessary though, and that the tree should handle putting the elements in the right order and handling empty causedBy arrays.